lib/base/string.hpp

```diff
@@ -205,4 +205,18 @@ struct range_const_iterator<icinga::String>
 
 }
 
+namespace std
+{
+
+template<>
+struct hash<::icinga::String>
+{
+	decltype(hash<string>{}(string())) operator()(const ::icinga::String& s) const noexcept
+	{
+		return hash<string>{}(s.GetData());
+	}
+};
+
+}
```

Comment:
Looks redundant to me: https://github.com/Icinga/icinga2/blob/dedd7d6928545b9b0b631c513ba5bc617149a8e7/lib/base/string.hpp#L183-L187